src/lauxlib.js

```diff
@@ -399,6 +400,124 @@ const luaL_newlib = function(L, l) {
     luaL_setfuncs(L, l, 0);
 };
 
+// Only with Node
+if (typeof window === "undefined") {
+    const fs = require('fs');
+
+    class LoadF {
+        constructor() {
+            this.n = NaN;  /* number of pre-read characters */
+            this.f = null;  /* file being read */
+            this.buff = new Buffer(1024);  /* area for reading file */
+            this.pos = 0;  /* current position in file */
+        }
+    }
+
+    const getF = function(L, ud) {
+        let lf = ud;
+        let bytes = 0;
+        if (lf.n > 0) {  /* are there pre-read characters to be read? */
+            lf.n = 0;  /* no more pre-read characters */
+        } else {  /* read a block from file */
+            lf.buff.fill(0);
+            bytes = fs.readSync(lf.f, lf.buff, 0, lf.buff.length, lf.pos); /* read block */
+            lf.pos += bytes;
+        }
+        return bytes > 0 ? new lobject.TValue(0, lf.buff).jsstring() : null; // TODO: Here reading utf8 only
+    };
+
+    const errfile = function(L, what, fnameindex, error) {
+        let serr = error.message;
+        let filename = lapi.lua_tostring(L, fnameindex).slice(1);
+        lapi.lua_pushstring(L, `cannot ${what} ${filename}: ${serr}`);
+        lapi.lua_remove(L, fnameindex);
+        return lua.thread_status.LUA_ERRFILE;
+    };
+
+    const getc = function(lf) {
+        let b = new Buffer(1);
+        let bytes = fs.readSync(lf.f, b, 0, 1, lf.pos);
+        lf.pos += bytes;
+        return bytes > 0 ? b.readUInt8() : null;
+    };
+
+    const skipBOM = function(lf) {
+        let p = "\xEF\xBB\xBF";  /* UTF-8 BOM mark */
```

Comment:
I'm not sure this will work correctly in JS? ==> Could you add a test that has a UTF-8 BOM?